src/openrct2-ui/windows/Ride.cpp

```diff
@@ -6810,6 +6818,19 @@ void WindowRideInvalidateVehicle(const Vehicle& vehicle)
     w->Invalidate();
 }
 
+void WindowRidePaintResetVehicle(RideId rideIndex)
+{
+    auto w = static_cast<RideWindow*>(WindowFindByNumber(WindowClass::Ride, rideIndex.ToUnderlying()));
+    if (w != nullptr)
+    {
+        if (w->page == 4) // WINDOW_RIDE_PAGE_COLOUR
+        {
+            w->ResetVehicleIndex();
+        }
+        w->Invalidate();
```

Comment:
Shouldn't this be inside the above if-statement, since otherwise the state is unchanged anyway?

Reply:
I'm not sure if that is strictly true. There might be other state that has been updated I'm unsure as to what the callers of this are expecting so I've just kept it as is when moving it over.

Reply:
Something to investigate afterwards in my opinion, it won't really make any difference at this point.